### PR TITLE
Sync support

### PR DESCRIPTION
**Pull request by acbramley**
Provides integration for
https://github.com/universityofadelaide/openshift-client/pull/52
https://github.com/universityofadelaide/shepherd-operator/pull/15

### CODE REVIEW

File: web/modules/custom/shepherd/shp_backup/src/Service/Backup.php [hunk 1]
```diff
@@ -97,13 +118,57 @@ public function createBackup(NodeInterface $site, NodeInterface $environment, st
...
+      'field_shp_env_vars' => $environment->field_shp_env_vars->getValue(),
+      'field_shp_cpu_limit' => $environment->field_shp_cpu_limit->value,
+      'field_shp_cpu_request' => $environment->field_shp_cpu_request->value,
+      'field_shp_memory_limit' => $environment->field_shp_memory_limit->value,
+      'field_shp_memory_request' => $environment->field_shp_memory_request->value,
+      'field_max_replicas' => $environment->field_max_replicas->value,
+      'field_min_replicas' => $environment->field_min_replicas->value,
+      'field_shp_secrets' => $environment->field_shp_secrets->value,
+    ]);
+    $new_environment->save();
```
Comment (acbramley) [inline web/modules/custom/shepherd/shp_backup/src/Service/Backup.php:164]: @pingers this doesn't seem to trigger a job to prefill the db, I'll have a hunt through and see why

Comment (acbramley) [reply web/modules/custom/shepherd/shp_backup/src/Service/Backup.php:164]: Ah, the project probably doesn't have a default sql file :)

**Comment by pingers** [comment]
:shipit:

File: web/modules/custom/shepherd/shp_backup/src/Service/Backup.php [hunk 2]
```diff
@@ -147,6 +214,18 @@ public function operationsLinksAlter(array &$operations, EntityInterface $entity
         'attributes' => $modal_attributes,
       ];
     }
+    // Allow upgrade on prod environments.
+    else {
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_backup/src/Service/Backup.php:218]: Remove the else to allow any env to upgrade.

Comment (pingers) [reply web/modules/custom/shepherd/shp_backup/src/Service/Backup.php]: Fixed.

**Pull request merged**